bchain/coins/eth/ethrpc.go

```diff
@@ -1219,20 +1219,26 @@ func (b *EthereumRPC) EthereumTypeGetNonce(addrDesc bchain.AddressDescriptor) (u
 			addrDesc,
 			"pending",
 		)
-		if err != nil {
+		if err == nil && result != "" {
+			useAlternative = true
+			glog.Errorf("Alternative provider success for eth_getTransactionCount")
+		} else {
 			glog.Errorf("Alternative provider failed for eth_getTransactionCount: %v, falling back to primary RPC", err)
 		}
 	}
 
-	if result == "" {
+	if !useAlternative {
 		result, err = b.callRpcStringResult("eth_getTransactionCount", addrDesc, "pending")
 		if err != nil {
+			glog.Errorf("Primary RPC failed for eth_getTransactionCount: %v", err)
 			return 0, err
 		}
+		glog.Errorf("Primary RPC result: '%s'", result)
```

Comment:
This logs success as error and like the other comment, I would not merge it to master as it will fill log with a lot of unnecessary lines.